react_native_fundamentals/presentation.html

```diff
@@ -2,43 +2,1137 @@
+
+•TextInput:
+A foundational component for inputting text into the app via a keyboard. Props provide configurability for several features, such as auto-correction, auto-capitalization, placeholder text, and different keyboard types, such as a numeric keypad. You can subscribe to the onChangeText events to read the user input. Other important props are the value and the placeholder. value: The value to show for the text input. TextInput is a controlled component, which means the native value will be forced to match this value prop if provided. Placeholder: 
+the string that will be rendered before text input has been entered. 
+•ScrollView:
+Component that wraps platform ScrollView while providing integration with touch locking "responder" system.
+Keep in mind that ScrollViews must have a bounded height in order to work, since they contain unbounded-height children into a bounded container (via a scroll interaction). In order to bound the height of a ScrollView, either set the height of the view directly (discouraged) or make sure all parent views have bounded height. 
+          </aside>
+        </section>
+
+        <section class="compare-slide">
+          <h4 class="slide-name">Using List Views</h4>
+          <div>
+          <div>
+            <img src="assets/images/flat-list.gif" alt="simple" />
+            <ul>
+              <li>long lists of data</li>
+              <li>renders items lazily</li>
+            </ul>
+          </div>
+          <div>
+            <img src="assets/images/scroll-view.gif" alt="simple" />
+            <ul>
+              <li>small number of things of a limited size</li>
+              <li>renders all its child components at once</li>
+            </ul>
+          </div>
+        </div>
+          <aside class="notes">
+            React Native provides a suite of components for presenting lists of data. Generally, there are two methods of listing data in a React Native app: displaying the list with map and using the FlatList component.
+The map method technically gives you more flexibility to customize your list the way you want with the React Native ScrollView component but it loads all the items every time which can be expensive.
+FlatList is a React Native component that only loads items that are currently visible on the screen and it deletes items as they go off screen which is more optimal for large amounts of data. Furthermore, it provides scrolling features by default. You don’t need to use extra components like ScrollView to make list data scrollable. The FlatList component requires two props: data and renderItem. data is the source of information for the list. renderItem takes one item from the source and returns a formatted component to render.
+If you need section support, you can use SectionList.
+          </aside>
+        </section>
+
+        <section class="several-pictures-4-slide-with-description">
+          <h4 class="slide-name">Touchables</h4>
```

Comment:
could you change the order of the pictures on the slide as they are mentioned in the notes or change the order of the components in the notes so that they match the pictures?

Reply:
Done